src/ranged.cpp

```diff
@@ -1449,7 +1449,10 @@ static void cycle_action( item& weap, const tripoint &pos ) {
     const auto mag = weap.magazine_current();
     if( mag && mag->type->magazine->linkage != "NULL" ) {
         item linkage( mag->type->magazine->linkage, calendar::turn, 1 );
-        if( cargo.empty() ) {
+        if (weap.gunmod_find("brass_catcher")) {
+            g->u.inv.add_item(linkage, true, false);
```

Comment:
If an NPC fires a gun that has a brass catcher installed, it will magically teleport the linkage into the inventory of the player character.

Reply:
crap, I'll push a follow-up PR